src/bitops.c

```diff
@@ -802,9 +802,9 @@ void bitcountCommand(client *c) {
     int isbit = 0;
     unsigned char first_byte_neg_mask = 0, last_byte_neg_mask = 0;
 
-    /* Lookup, check for type, and return 0 for non existing keys. */
-    if ((o = lookupKeyReadOrReply(c,c->argv[1],shared.czero)) == NULL ||
-        checkType(c,o,OBJ_STRING)) return;
+    /* Lookup, check for type. */
+    o = lookupKeyRead(c->db, c->argv[1]);
```

Comment:
i think we should move the lookupKey call to after the argument parsing.
we don't want to do the checkType or even update the lookup stats when there's a syntax error

Reply:
I thought about this, but then I recall this one: https://github.com/redis/redis/pull/9271/files#diff-3a1ab5e27220b022425b8cf51decd6e1ecdd0366dfef7569383771e3577154c3R521

Reply:
what about it?

Reply:
we call lookupKey before the argument parsing

Reply:
well, i suppose that's wrong there as well.
in that other PR it was also the case of fixing one problem and skipping the other.
was there any discussion there about moving it which resulted in a realization that we shouldn't (as far as i can tell it wasn't discussed).

maybe we're looking too much into how a command with invalid arguments behaves.
i'm certain it's better to lookup after argument parsing, but maybe moving things around in some complicated commands can cause complications and other issues.

so if it's complicated, i suppose that for now we can focus on fixing the return value, and not the other side effects.

Reply:
you are right, it wasn't discussed. and lookup after the argument parsing is better.
in this case, it is a bit complicated.  the reason is that:
```
p = getObjectReadOnlyString(o,&strlen,llbuf);
long long totlen = strlen;
if (isbit) totlen <<= 3;
if (start < 0) start = totlen+start;
if (end < 0) end = totlen+end;
if (start < 0) start = 0;
if (end < 0) end = 0;
if (end >= totlen) end = totlen-1;
```

there are some code below rely the `totlen`, and `totlen` rely on `strlen`, which rely on the robj (lookupKey)

so i think for now we can just fix the return value problem